File: trunk/src/java/org/codehaus/testdox/intellij/TestDoxInterface.java
package org.codehaus.testdox.intellij;

import javax.swing.Icon;

import com.intellij.openapi.vfs.VirtualFile;

import org.codehaus.testdox.intellij.panel.TestDoxModel;

public class TestDoxInterface extends TestDoxFile {

    public static final TestElement TEST_ELEMENT = new AbstractTestElement() {
        public String displayString() {
            return "<font color=\"gray\">Interfaces do not have unit tests.</font>";
        }

        public Icon icon() {
            return IconHelper.getIcon(IconHelper.WARNING_ICON);
        }
    };

    public TestDoxInterface(VirtualFile file, String className, TestClass testClass, TestClass testedClass) {
        super(file, className, testClass, testedClass, TestMethod.EMPTY_ARRAY);
    }

    public void updateModel(TestDoxModel model) {
        model.setTestDoxForInterface(this);
    }
}

File: src/scala/org/codehaus/testdox/intellij/TestDoxInterface.scala
package org.codehaus.testdox.intellij

import com.intellij.openapi.vfs.VirtualFile
import org.codehaus.testdox.intellij.ui.TestDoxTableModel

class TestDoxInterface(file: VirtualFile, className: String, testClass: TestClass, testedClass: TestClass)
    extends TestDoxFile(file, className, testClass, testedClass, TestMethod.EMPTY_ARRAY) {

  def updateModel(model: TestDoxTableModel) {
    model.setTestDoxForInterface(this)
  }
}

object TestDoxInterface {

  val TEST_ELEMENT = new AbstractTestElement() {
    val displayString = "<font color=\"gray\">Interfaces do not have unit tests.</font>"
    val icon = Icons.getIcon(Icons.WARNING_ICON)
  }
}
